internal/pkg/util/fs/overlay/overlay_item_linux.go

```diff
@@ -0,0 +1,297 @@
+// Copyright (c) 2019-2023, Sylabs Inc. All rights reserved.
+// This software is licensed under a 3-clause BSD license. Please consult the
+// LICENSE.md file distributed with the sources of this project regarding your
+// rights to use or distribute this software.
+
+package overlay
+
+import (
+	"fmt"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"strings"
+	"syscall"
+
+	"github.com/sylabs/singularity/internal/pkg/util/bin"
+	"github.com/sylabs/singularity/pkg/image"
+	"github.com/sylabs/singularity/pkg/sylog"
+)
+
+// Item represents information about a single overlay item (as specified,
+// for example, in a single --overlay argument)
+type Item struct {
+	// Type represents what type of overlay this is (from among the values in pkg/image)
+	Type int
+
+	// Writable represents whether this is a writable overlay
+	Writable bool
+
+	// BarePath is the path of the overlay stripped of any colon-prefixed
+	// options (like ":ro")
+	BarePath string
+
+	// DirToMount is the path of the directory that will actually be passed to
+	// the mount system-call when mounting this overlay
+	DirToMount string
```

Comment:
The name `DirToMount` and the comment are a bit confusing to me. The value is the directory from which this `Item` should be overlayfs mounted in the `Set` context. However, we may have mounts in the `Item` context, so we need to be very specific when we talk about mounts.

I'd suggest something like `StagingDir`, but mostly want to encourage emphasizing that the mount you are talking about is in the Set context, something more like `... is the directory on which this overlay item is staged, to be used as a source for an overlayfs mount as part of an overlay.Set`